pkg/conference/participant.go

```diff
@@ -64,3 +64,77 @@ func (p *Participant) sendDataChannelMessage(toSend event.Event) error {
 
 	return nil
 }
+
+type TrackMetadata struct {
+	maxWidth, maxHeight int
+}
+
+func (t TrackMetadata) fullResolution() int {
+	return t.maxWidth * t.maxHeight
+}
+
+func (t TrackMetadata) isVideoTrack() bool {
+	return t.fullResolution() > 0
+}
+
+type PublishedTrackInfo struct {
+	peer.TrackInfo
+	availableLayers []peer.SimulcastLayer
+	metadata        TrackMetadata
+}
+
+func (p *Participant) getPublishedTracksInfo() map[string]PublishedTrackInfo {
+	tracksMetadata := make(map[string]TrackMetadata)
+	for _, metadata := range p.streamMetadata {
+		for id, track := range metadata.Tracks {
+			tracksMetadata[id] = TrackMetadata{
+				maxWidth:  track.Width,
+				maxHeight: track.Height,
+			}
+		}
+	}
+
+	publishedTracksMetadata := make(map[string]PublishedTrackInfo)
+	for id, track := range p.publishedTracks {
+		metadata, ok := tracksMetadata[id]
+
+		if !ok {
+			// We don't have metadata for this track, so we can't send it to the client.
+			p.logger.Warnf("No metadata for published track %s", id)
+			continue
+		}
+
+		publishedTracksMetadata[id] = PublishedTrackInfo{
+			TrackInfo:       track.info,
+			availableLayers: track.layers,
+			metadata:        metadata,
+		}
+	}
+
+	return publishedTracksMetadata
+}
+
+func (p *PublishedTrackInfo) getDesiredLayer(requestedWidth, requestedHeight int) *peer.SimulcastLayer {
+	// Audio track. For them we don't have any simulcast.
+	if p.metadata.isVideoTrack() {
```

Comment:
```suggestion
	if !p.metadata.isVideoTrack() {
```

Reply:
Yep, was indeed a bug in the initial commit, [fixed it by accident in this commit](https://github.com/matrix-org/waterfall/pull/85/commits/59579be99a6879efbb6a342d2ed864d5147f596b).